doc/geom/SpanSet.rst

```diff
@@ -0,0 +1,181 @@
+=======
+SpanSet
+=======
+A ``SpanSet`` is a class that represents regions of pixels on an image.  As the
+name implies, a ``SpanSet`` is a collection of ``Span`` objects which behave as
+a mathematical set. The class provides methods for doing set operations, as well
+as convenience method for operating on data products at the locations defined
+by the ``SpanSet``. The following example shows a few different ways to create
+a ``SpanSet`` in python.
+
+.. code-block:: python
+
+    from lsst.afw.geom import Span, SpanSet, Stencil
+
+    # Construct a SpanSet from a list of Spans
+    spanList = [Span(1, 0, 2), Span(2, 0, 2), Span(3, 0, 2)]
+    spanSetFromList = SpanSet(spanList)
+
+    # Construct the same SpanSet from a shape
+    spanSetFromShape = SpanSet.fromShape(1, Stencil.BOX, offset=(1, 2))
+
+    # Verify that the two are the same
+    assert(spanSetFromList == spanSetFromShape)
+
+The last example introduced a useful factory method for creating ``SpanSets``
+and is worth exploring further. ``fromShape`` can be called in one of two ways,
+specifying a radius, stencil (defaults to a circle), and an offset (defaults to
+0, 0), or passing in an ``afw.geom.ellipse.Ellipse`` object. There are three
+options when creating a ``SpanSet`` from a ``afw.geom.Stencil``; ``BOX``,
+``CIRCLE``, and ``MANHATTAN`` (a diamond). Each shape is centered on the 0, 0
+pixel and is extended the given radius in all directions according to the shape
+selected. If an origin other than 0, 0 is desired an x, y offset to the origin
+can be provided.
+
+When working with ``SpanSet`` objects it is important to note that they are
+immutable once they are created. As such, all method calls on a ``SpanSet``
+object which operate on a ``SpanSet`` create, modify and return a new
+``SpanSet`` object. This design is intended to offer a safety guarantee to
+programmers. If two different bits of code each have a shared pointer to the
+same ``SpanSet`` one can not modify the region and possible corrupt what the
+other bit of code is trying to do.
+
+Another important note is that regions defined by a ``SpanSet`` are not
+guaranteed to be contiguous. There is however a method which will return if a
+given ``SpanSet`` is contiguous, and a method which will split an
+non-contiguous ``SpanSet`` into a standard vector (or list in python) of
+contiguous ``SpanSet`` objects. The following example demonstrates this behavior
+along side set operations.
+
+.. code-block:: python
+
+    from lsst.afw.geom import SpanSet, Stencil
+
+    # Construct some SpanSets
+    ss1 = SpanSet.fromShape(5)
+    ss2 = SpanSet.fromShape(5, offset=(6, 6))
+
+    # Create a union of the immutable sets
+    ss3 = ss1.union(ss2)
+
+    # Erode the combine SpanSet with a circular kernel and check if the result
+    # is contiguous
+    ssEroded = ss3.eroded(4, Stencil.CIRCLE)
```

Comment:
(Also, and this is probably obvious, but just to draw @natelust's attention to it — Jonathan has made some useful comments here, but, since you've pushed other changes, GitHub is now hiding them behind "outdated diff" notices. It's worth going through and checking which are still relevant: the ones I looked at certainly were. Sorry if you were planning to do this anyway!)